### PR TITLE
test: mempool : improve coverage to mempool

### PR DESCRIPTION
**Pull request by yingmingx**
Add some testcase to improve coverage of mempool.c

Signed-off-by: Ying ming <mingx.ying@intel.com>

### CODE REVIEW

File: tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		if (_is_user_context()) {
+			k_oops();
+		}
+#endif
+		k_panic();
+	}
+}
+
+/**
+ * @brief Test Test input aligen value zero
```
Comment (LeiW000) [inline tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c:56]: Double "Test". Please remove one.

File: tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+{
+	set_assert_valid(true);
+
+	k_aligned_alloc(0, 1);
+
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test Test input aligen value 5
```
Comment (LeiW000) [inline tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c:72]: Double "Test". Please remove one.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c (hunk 1 above)
Comment (yingmingx) [reply tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c]: I've changed it. Thank you for your careful review.

**Review by LeiW000 — APPROVED** [review]

**Review by enjiamai — CHANGES_REQUESTED** [review]
The common fatal and assert handler has been merged, could you please leverage them instead? thanks!  
please refer to https://github.com/zephyrproject-rtos/zephyr/pull/30150

**Comment by yingmingx** [comment]
> The common fatal and assert handler has been merged, could you please leverage them instead? thanks!
> please refer to #30150

Hi, enjia. All the assert related code have been removed and fatal framework have been used. Please help to give another look, thanks.

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c [hunk 3]
```diff
@@ -118,3 +162,55 @@ void test_k_aligned_alloc(void)
 		"%p not 16-byte-aligned", r);
 	k_free(r);
 }
+
+/**
+ * @brief Validate allocation and free from system heap memory pool
+ *
+ * @ingroup kernel_heap_tests
+ *
```
Comment (enjiamai) [inline tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c:170]: please add the @details to descript the high level testing purpose or concept, thanks!

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c [hunk 4]
```diff
@@ -118,3 +162,55 @@ void test_k_aligned_alloc(void)
...
+	k_free(ptr);
+
+	zassert_is_null((char *)z_thread_malloc(BLK_SIZE_MAX * 2),
+						"overflow check failed");
+}
+
+/**
+ * @brief Validate allocation and free from system heap memory pool
+ * in isr context
+ *
```
Comment (enjiamai) [inline tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c:189]: same as line 170

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c [hunk 5]
```diff
@@ -118,3 +162,55 @@ void test_k_aligned_alloc(void)
...
+ */
+void test_malloc_in_isr(void)
+{
+	irq_offload((irq_offload_routine_t)tIsr_malloc_and_free, NULL);
+}
+
+/**
+ * @brief Validate allocation and free failure when thread's resource pool
+ * is not assigned
+ *
```
Comment (enjiamai) [inline tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c:202]: same as line 170

**Review by enjiamai — CHANGES_REQUESTED** [review]
could you please add the @details descriptions, thanks!

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c (hunk 3 above)
Comment (yingmingx) [reply tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c:174]: Hi, enjia.  Details have been added. Thanks.

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c (hunk 4 above)
Comment (yingmingx) [reply tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c:197]: Hi, enjia. Details have been added. Thanks.

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c (hunk 5 above)
Comment (yingmingx) [reply tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_api.c:212]: Hi, enjia. Details have been added. Thanks.

**Review by enjiamai — APPROVED** [review]

**Comment by yingmingx** [comment]
Close this PR, the new PR is #32767.

**Pull request closed without merging**